File: packages/styles/src/templates/flutter/component_styles.dart
part of <%= packageName %>;
<%
var getValuePrefix = ({attributes}) => {
  const { subitem } = attributes;
  console.log(subitem)
  if(subitem === "backgroundColor" || subitem === "color") {
    return "const ";
  }
  return "";
}
_.forEach(props, (prop) => { const className = _.upperFirst(prop.key) + "Styles" %>
class <%- className %> {
  <%- className %>._();
<% _.forEach(prop.value, (componentStyle) => {%>
<%
if(componentStyle.comment) { %>  /// <%= componentStyle.comment %><%}
else {
%>  /// <%= componentStyle.attributes.subitem %> for <%= componentStyle.attributes.item %><% } %>
  static const <%= componentStyle.name %> = <%= getValuePrefix(componentStyle) %><%= componentStyle.value %>;
<% }); %>}
<% }); %>

File: packages/style-dictionary/src/templates/web/component_styles.js
/* eslint-disable max-lines,sort-keys */

<%
_.forEach(props, (prop) => {
%>export const <%=prop.key%>Styles = {
<%
const classKeys = getAllPropertiesByItem(prop.value);
_.forEach(classKeys, (classKey) => { %>
  <%= classKey.key %>: {<%
_.forEach(classKey.value, (property) => {
  const { attributes, comment, value } = property;
  const { item, subitem, type } = attributes;
  const name = getCssVarName({ classKey: item, component: type, property: subitem });
  const fallbackComment = `${_.startCase(subitem)} for ${_.startCase(type)} ${_.startCase(item)}`;
%>

    /**
     * <%= comment ? comment : fallbackComment %>
     */
    <%= subitem %>: "var(<%= name %>, <%= value %>)",
<% }); %>
  },
<% }); %>
};

<% }); %>
